### PR TITLE
Closes #17382. Add `\yii\validators\DateValidator::$strictDateFormat`…

### LINKED ISSUE #17382: Date validator is not working properly
### What steps will reproduce the problem?
Input Date: 13-Mar-19
System Format: medium `($this->format used in DateValidator)`
 
When `timestampAttribute` set that time not working. like
```
['approval_date', 'date', 'timestampAttribute' => 'approval_date', 'timestampAttributeFormat' => 'yyyy-MM-dd'],
// That Consider a date is valid but is not a valid
```
```
['approval_date', 'date']
//Working Properly
```

### What is the expected result?
Need invalid date message when configure `timestampAttribute` and `timestampAttributeFormat`.

### What do you get instead?
Date is validated

### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.17
| PHP version      | 7.2
| Operating system | Ubuntu 18.04

### PR DESCRIPTION
**Pull request by alexkart**
… to enable strict validation.

| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #17382

### CODE REVIEW

**Comment by alexkart** [comment]
@samdark what do you think?

File: framework/validators/DateValidator.php [hunk 1]
```diff
@@ -364,35 +373,80 @@ private function parseDateValueFormat($value, $format)
...
+     * Performs strict date format validation
+     *
+     * @param $value string date
+     * @param $parsedDate string parsed date
+     * @param IntlDateFormatter $formatter formatter
+     * @return bool
+     */
+    private function validateStrictIntl($value, $parsedDate, IntlDateFormatter $formatter)
+    {
+        if ($this->strictDateFormat) {
```
Comment (samdark) [inline framework/validators/DateValidator.php]: How about moving this condition out of the method?

Comment (samdark) [reply framework/validators/DateValidator.php]: I'm suggesting it because method itself implies that validation is going to be strict.

**Review by samdark — APPROVED** [review]
The approach seems to be OK.

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
Merged. Thank you!

**Pull request merged**